feat: nested folders

eliezer-castro opened the pull request:
This PR refactors the FoldersPane component to replace the flat list layout with a tree view using react-arborist. It introduces support for nested folders, enabling better folder hierarchy visualization and interaction.


https://github.com/user-attachments/assets/dac90284-7014-4357-ad33-30cef0653cf3

kimatata commented:
Hi, 

Thanks again for this awesome PR! 🎉
I really appreciate you picking up the nested folders issue — I had been putting that off for a while, so this is super helpful.

I’ve tested the branch and did a quick review. Overall, great work — the integration with react-arborist looks clean and fits really well into the existing UI. Just a couple of things I’d like to ask before merging:

## runs page is still flat:
It doesn’t yet reflect the nested folder structure, but it’s actually okay. All folders are still shown (flat). 
You can create a separate issue for this — no need to handle it in this PR.

<img width="1195" height="838" alt="image" src="https://github.com/user-attachments/assets/90598ae7-bc68-49e3-a0b7-c6305f440075" />

## Child folders not deleted when parent is removed:

If a parent folder with children is deleted, the children stay in the DB.
I think this can be fixed by adding a foreign key constraint in `backend/models/folder.js` like this:

```js
parentFolderId: {
  type: Sequelize.INTEGER,
  allowNull: true,
  references: {
    model: 'folders',
    key: 'id',
  },
  onDelete: 'CASCADE',
  onUpdate: 'CASCADE',
},
```

This way, deleting a folder will automatically remove its children.

## Small Type Improvement:

The `TreeNodeData` interface currently defined inline could be moved to `frontend/types/folder.ts` to avoid duplication.
Also, I think `detail?` should just be `detail`, since the column is always present.

Thanks again for your great contributions 🙌

eliezer-castro commented:
Thanks a lot for the thoughtful review and feedback! 🙌

I’ve just pushed the requested changes:

Cascade delete: added onDelete: 'CASCADE' and onUpdate: 'CASCADE' to the parentFolderId field in the model — child folders are now removed when their parent is deleted.

Types: moved the TreeNodeData interface to frontend/types/folder.ts and updated references. Also changed detail? to detail to match the DB schema.

Runs page structure: I went ahead and updated the runs page to also support the nested folder structure. I introduced a reusable TreeItem component, which is now shared between the test cases and executions views.

Let me know if there’s anything else you’d like to tweak!
